blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -108,21 +128,53 @@ public BluePipeline create(Reachable parent) throws IOException {
                                         ));
                     }
                 }
-                GitHubSCMNavigator gitHubSCMNavigator = new GitHubSCMNavigator(apiUrl, orgName, credentialId, credentialId);
-                if (sb.length() > 0) {
-                    gitHubSCMNavigator.setPattern(sb.toString());
-                }
 
                 // cick of github scan build
                 OrganizationFolder organizationFolder = (OrganizationFolder) item;
+
+                GitHubSCMNavigator gitHubSCMNavigator = organizationFolder.getNavigators().get(GitHubSCMNavigator.class);
+
+                StringBuilder sb = new StringBuilder();
+                if (gitHubSCMNavigator != null) {
+                    Matcher matcher = Pattern.compile("\\((.*?)\\\\b\\)\\?").matcher(gitHubSCMNavigator.getPattern());
```

Comment:
This might benefit from a comment - it's clear to me because I'm familiar w/ how the repo names are put in the regexp, but others might be confused by this.